File: remove_bg.php
<?php
$inputFile = 'public/nuevo icono.png';
$outputFile = 'public/nuevo_icono_transparent.png';
$tolerance = 220; // 0-255, what counts as "white"

if (!file_exists($inputFile)) {
    die("Error: File not found.\n");
}

$img = imagecreatefrompng($inputFile);
if (!$img) {
    die("Error: Could not load PNG.\n");
}

// Enable alpha blending
imagealphablending($img, false);
imagesavealpha($img, true);

$width = imagesx($img);
$height = imagesy($img);

for ($x = 0; $x < $width; $x++) {
    for ($y = 0; $y < $height; $y++) {
        $color = imagecolorat($img, $x, $y);
        $rgba = imagecolorsforindex($img, $color);
        
        // If it's white or near-white
        if ($rgba['red'] > $tolerance && $rgba['green'] > $tolerance && $rgba['blue'] > $tolerance) {
            // Make it fully transparent
            $transparent = imagecolorallocatealpha($img, 0, 0, 0, 127);
            imagesetpixel($img, $x, $y, $transparent);
        }
    }
}

if (imagepng($img, $outputFile)) {
    echo "Success: Saved to $outputFile\n";
} else {
    echo "Error: Could not save PNG.\n";
}
imagedestroy($img);
?>

File: remove_bg.py
from PIL import Image
import numpy as np
import sys

def remove_white_bg(input_path, output_path, tolerance=220):
    try:
        img = Image.open(input_path).convert("RGBA")
        data = np.array(img)
        
        # Get r, g, b, a channels
        r, g, b, a = data.T
        
        # Define the white areas (R, G, B all > tolerance)
        white_areas = (r >= tolerance) & (g >= tolerance) & (b >= tolerance)
        
        # Set alpha to 0 for white areas
        data[..., 3][white_areas.T] = 0
        
        new_img = Image.fromarray(data)
        new_img.save(output_path, "PNG")
        print(f"Success: Saved to {output_path}")
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    remove_white_bg('public/nuevo icono.png', 'public/nuevo_icono_transparent.png')
